### PR TITLE
CLDR-14057 Survey Tool CLDR_PHASE=FINAL_TESTING should not allow vetters to vote

### PR DESCRIPTION
**Pull request by btangmu**
-New function SurveyMain.isPhaseFinalTesting called by userCanModifyLocaleWhy and manage.jsp

-New message ModifyDenial.DENY_PHASE_FINAL_TESTING

-Clean up formatting of enum ModifyDenial

-Fix compiler warnings about static SurveyMain.QUERY_DO and SurveyMain.getNewVersion

-Fix compiler warning about UserRegistry.writeUserFile, call out.close before return

-Remove superfluously duplicated CLDR_PHASE setting in cldr-properties.j2

-Delete some dead code

##### Checklist

- [x] Issue filed: https://unicode-org.atlassian.net/browse/CLDR-14057
- [x] Updated PR title and link in previous line to include Issue number

### CODE REVIEW

File: tools/cldr-apps/src/main/java/org/unicode/cldr/web/UserRegistry.java [hunk 1]
```diff
@@ -1760,10 +1760,15 @@ static boolean localeMatchesLocaleList(String localeList, CLDRLocale locale) {
...
-                    "User does not have any voting rights"), DENY_LOCALE_LIST("User does not have rights to vote for this locale");
+        DENY_NULL_USER("No user specified"),
+        DENY_LOCALE_READONLY("Locale is read-only"),
+        DENY_PHASE_READONLY("SurveyTool is in read-only mode"),
+        DENY_ALIASLOCALE("Locale is an alias"),
+        DENY_DEFAULTCONTENT("Locale is the Default Content for another locale"),
+        DENY_PHASE_CLOSED("SurveyTool is in 'closed' phase"),
+        DENY_NO_RIGHTS("User does not have any voting rights"),
+        DENY_LOCALE_LIST("User does not have rights to vote for this locale"),
+        DENY_PHASE_FINAL_TESTING("SurveyTool is in the 'final testing' phase");
```
Comment (btangmu) [inline tools/cldr-apps/src/main/java/org/unicode/cldr/web/UserRegistry.java:1771]: DENY_PHASE_FINAL_TESTING is new; the rest is just reformatted

File: tools/cldr-apps/src/main/java/org/unicode/cldr/web/UserRegistry.java [hunk 2]
```diff
@@ -2453,6 +2461,7 @@ int writeUserFile(SurveyMain sm, String ourDate, boolean obscured, File outFile)
                 rs = ps.executeQuery();
                 if (rs == null) {
                     out.println("\t<!-- No results -->");
+                    out.close();
```
Comment (btangmu) [inline tools/cldr-apps/src/main/java/org/unicode/cldr/web/UserRegistry.java:2464]: This fixes a compiler warning -- unrelated to this ticket but same file and too small for its own ticket

File: tools/cldr-apps/src/main/webapp/WEB-INF/tmpl/manage.jsp [hunk 3]
```diff
@@ -55,9 +52,11 @@ String helpName = subCtx.getString("helpName");
                 subCtx.println("<br> (Note: in the Vetting phase, you may not submit new data.) ");
             } else if(SurveyMain.isPhaseClosed() && !UserRegistry.userIsTC(subCtx.session.user)) {
                 subCtx.println("<br>(SurveyTool is closed to vetting and data submissions.)");
+            } else if (SurveyMain.isPhaseFinalTesting() && !UserRegistry.userIsTC(subCtx.session.user)) {
+                subCtx.println("(SurveyTool is in the Final Testing phase; you may not vote.)");
             }
```
Comment (btangmu) [inline tools/cldr-apps/src/main/webapp/WEB-INF/tmpl/manage.jsp:57]: This is the most essential addition. The other changes in this file fix compiler warnings or delete dead ("if false") code

File: tools/scripts/ansible/templates/cldr-properties.j2 [hunk 4]
```diff
@@ -22,9 +22,6 @@ CLDR_OLDVERSION={{ surveytooldeploy.oldversion }}
 ## 'new'  version
 CLDR_NEWVERSION={{ surveytooldeploy.newversion }}
 
-## Current SurveyTool phase 
-CLDR_PHASE=BETA
-
 ## CLDR trunk. Default value shown
```
Comment (btangmu) [inline tools/scripts/ansible/templates/cldr-properties.j2:25]: Somehow this template had "CLDR_PHASE=BETA" twice but only needs it once

**Review by macchiati — APPROVED** [review]

**Pull request merged**